v5/v5_with_library/v5_stable.ino

```diff
@@ -617,49 +618,53 @@ String getResponse() {
     response += SerialAT.readString();
   }
   Println(3, "After while loop in get response");
-  if (response.indexOf("+CMTI:") != -1) { // new message
+  if (response.indexOf("+CMTI:") != -1 || pending_test_message) { // new message
+    Println(3, "new message content :{" + response + "}");
+    int newMessageNumber;
+    String senderNumber, _message_;
     messages_in_inbox++;
-    int newMessageNumber = getNewMessageNumber(response);
-    String senderNumber = getMobileNumberOfMsg(String(newMessageNumber), true);
-    String _message_ = (removeNewline(removeOk(readMessage(newMessageNumber))));
-    if (senderNumber.indexOf("3374888420") == -1) {
+    if (pending_test_message) {
+      // pending_test_message = false;
+      newMessageNumber = demo_message_index.toInt();
+      senderNumber = demo_message_number;
+      _message_ = demo_message_content;
+    } else {
+      newMessageNumber = getNewMessageNumber(response);
+      senderNumber = getMobileNumberOfMsg(String(newMessageNumber), true);
+      _message_ = (removeNewline(removeOk(readMessage(newMessageNumber))));
+    }
+    if (!isIn(senderNumber, "3374888420")) {
```

Comment:
I suggest this string should also move to config or SPI memory.

Reply:
But you can mark it as resolve as I'm already late in merging. But please move this string to config or SPI.